apps/api/src/domain/models/term/tokenization/ChilcotinAlphabetParser.ts

```diff
@@ -0,0 +1,259 @@
+
+        this.root.registerTransition('d', d);
+
+        // 7 from t
+        const t = new Node('t')
+            .registerTransition('’', new Node('t’'))
+            .registerTransition('s', new Node('ts').registerTransition('’', new Node('ts’')))
+            .registerTransition('ŝ', new Node('tŝ').registerTransition('’', new Node('tŝ’')))
+            .registerTransition('l', new Node('tl').registerTransition('’', new Node('tl’')));
+
+        this.root.registerTransition('t', t);
+
+        // 4 from k
+        const k = new Node('k')
+            .registerTransition('w', new Node('kw').registerTransition('’', new Node('kw’')))
+            .registerTransition('’', new Node('k’'));
+
+        this.root.registerTransition('k', k);
+
+        // 4 from q
+        const q = new Node('q')
+            .registerTransition('w', new Node('qw').registerTransition('’', new Node('qw’')))
+            .registerTransition('’', new Node('q’'));
+
+        this.root.registerTransition('q', q);
+
+        // 5 from g
+        const g = new Node('g')
+            .registerTransition('w', new Node('gw'))
+            .registerTransition('g', new Node('gg').registerTransition('w', new Node('ggw')))
+            .registerTransition('h', new Node('gh'));
+
+        this.root.registerTransition('g', g);
+
+        // 2 from l
+        const l = new Node('l').registerTransition('h', new Node('lh'));
+
+        this.root.registerTransition('l', l);
+
+        // 2 from x
+        const x = new Node('x').registerTransition('w', new Node('xw'));
+
+        this.root.registerTransition('x', x);
+
+        // 2 from s
+        const s = new Node('s').registerTransition('h', new Node('sh'));
+
+        this.root.registerTransition('s', s);
+
+        // 2 from w
+        const w = new Node('w').registerTransition('h', new Node('wh'));
+
+        this.root.registerTransition('w', w);
+
+        // 2 from c (c doesn't count, as it isn't a valid letter on its own)
+        const c = new Node('c', false).registerTransition(
+            'h',
+            new Node('ch').registerTransition('’', new Node('ch’'))
+        );
+
+        this.root.registerTransition('c', c);
+
+        // 18 isolated letters that have no valid transitions
+        isolatedLetters.forEach((l) => this.root.registerTransition(l, new Node(l)));
+
+        const letters = this.root.toList();
+
+        const uniqueLetters = Array.from(new Set(letters));
+
+        // check for duplicates
+        assert(letters.length === uniqueLetters.length);
+
+        const NUMBER_OF_LETTERS_IN_ALPHABET = 53;
+
+        const actualSize = uniqueLetters.length;
+
+        assert(actualSize === NUMBER_OF_LETTERS_IN_ALPHABET);
+    }
+
+    size(): number {
+        const letters = this.root.toList();
+
+        return letters.length;
+    }
+
+    listAll(): string[] {
+        return this.root.toList();
+    }
+
+    parse(input: string): AlphabetCharacters[] {
```

Comment:
We should make a note somewhere that punctuation is currently treated as out-of-alphabet.